src/app/Shared/Services/Report.service.tsx

```diff
@@ -43,7 +52,13 @@ export class ReportService {
       headers,
     }).pipe(
       concatMap((resp) => {
+        // 200 indicates that the backend has the report cached and it will return
+        // the json in the response.
         if (resp.ok) {
+          if (resp.status == 202) {
+            this.notifications.info('Report generation in progress', 'Report is being generated');
+            resp.text().then((value) => this.jobIds.add(value));
+          }
```

Comment:
In this case, we emit a notification to inform the user about the async generation, and we start tracking the job ID - but we still end up returning an `Observable<AnalysisResult[]>` to the calling component, which we try to resolve with a value immediately based on the current response status body.

I think what we need here is that the returned Observable has a resolved value immediately if the response from the Cryostat server was an immediate cached report. If the server responded with the status code indicating async generation, then the Observable that we return to the component should not resolve yet - it should resolve once the result is actually available. We'll know it's available when we get the notification from the server. Then I guess this service can re-fire a request to the server to retrieve the actual report response body, and then finally we can resolve the Observable that we returned to the component.